src/kernels/gpu_reference_kernel/naive_conv.cpp

```diff
@@ -116,9 +139,13 @@ inline __device__ __host__ int8_t cast_to(const int32_t& val)
 /// at index 3 while stride for C is at index 0. This is different from how the
 /// tensor descriptors store strides, which is always NCHW order, left-to-right.
 
+/// alpha and beta are double to ensure high precision.
+
 template <bool ASSUME_PACKED, typename src_data_t, typename acc_data_t, typename dst_data_t>
 inline __device__ void naive_conv_fwd_nchw(const src_data_t* __restrict__ p_in,
                                            const src_data_t* __restrict__ p_wei,
+                                           const double alpha,
+                                           const double beta,
```

Comment:
kdb incompatibility

Reply:
@bghimireamd This is relatively easy to fix: Simply rename the kernel (and update the solver). That would lead to cache misses, but these are relatively harmless. You can even avoid cache misses if you keep the kernel with old signature, but this is a bit more work (and not worth it since the KDB needs to be regenerated anyway -- to cache new kernels).

Reply:
@atamazov : how does this change affect KDB? I'm missing some details in my understanding.

Reply:
@amberhassaan We supply "precompiled kernels package" which actually contains KDB files. The users can download and install the package on their systems. After this PR, KDB becomes outdated, but MIOpen will try to use it. And if it try to load and launch use outdated `naive_conv_fwd_nchw`, then HIP runtime will fail to do so because of mismatch of the number of parameters.

[Informative] IIRC, QA engineers use precompiled kernels packages during some testing rounds.

The per-user binary cache (UKDB)  is not affected because pathnames of UKDB files contain MIOpen version including git hash.